docs/internal/docs.rst

```diff
@@ -215,6 +216,58 @@ These come in the form of `RST roles <https://docutils.sourceforge.io/docs/ref/r
 
     :omgissue:`this is **the issue** <DDSXTY14-29>`
 
+.. rst:role:: omgspec
+
+  The OMG specs are published as PDFs and it would be nice to be able to link to subsections within the specs using something like ``DDS.pdf#2.2.2``.
+  It's possible for the OMG to enable that when the PDF is created, but they don't.
+  This role works around that by downloading the PDFs and extracting sections from the table of contents to generate links that that can be used by browsers.
+  If a spec PDF can't be downloaded, the link will be just to the spec PDF.
+
+  .. code-block:: rst
+
+    :omgspec:`dds`
+
+    :omgspec:`dds:2.2.2`
+
+    :omgspec:`dds:Annex B`
+
+    :omgspec:`Somewhere in the XTypes spec <xtypes>`
+
+    :omgspec:`Exactly here in the XTypes spec <xtypes:2.2.2>`
+
+    :omgspec:`TypeObject IDL <xtypes:Annex B>`
+
+  Turns into:
+
+    :omgspec:`dds`
+
+    :omgspec:`dds:2.2.2`
+
+    :omgspec:`dds:Annex B`
+
+    :omgspec:`Somewhere in the XTypes spec <xtypes>`
+
+    :omgspec:`Exactly here in the XTypes spec <xtypes:2.2.2>`
+
+    :omgspec:`TypeObject IDL <xtypes:Annex B>`
+
+  The format of the target is ``SPEC[:SECTION]``.
+
+  Valid specs are:
+
+  .. omgspecs::
+
+  These are defined in :ghfile:`docs/conf.py`.
+
+  Valid sections can be named one of two ways:
+
+  - Section that start with section numbers, such as ``2.2.2``.
+    This is based on the start of the full name of the section in the table of contents.
+  - All others can be linked using the section title, either part or whole, such as ``Anex B`` or ``Annex B - Syntax for Queries and Filters``.
+    This can be used to link to sections that don't have section numbers, but can be ambiguous, so should be most or all of the title.
```

Comment:
```suggestion
    This can be used to link to sections that don't have section numbers, but can be ambiguous, so it should be most or all of the title.
```